libraries/botbuilder-core/src/transcriptLogger.ts

```diff
@@ -47,8 +47,24 @@ export class TranscriptLoggerMiddleware implements Middleware {
 
         // hook up onSend pipeline
         context.onSendActivities(async (ctx: TurnContext, activities: Partial<Activity>[], next2: () => Promise<ResourceResponse[]>) => {
-            // run full pipeline
-            const responses: ResourceResponse[] = await next2();
+            // Run full pipeline and create a pointer to the return value from the pipeline.
+            let nextValue: ResourceResponse[] = await next2();
+            let responses: ResourceResponse[] = nextValue;
+
+            // If the return value from the pipeline is not an Array, point `responses` to an array of empty ResourceResponses.
+            // This allows for the following:
+            //      1. Allows the transcript logging to progress and not error out.
+            //         See https://github.com/microsoft/botbuilder-js/issues/1404
+            //      1. This prevents TranscriptLoggers from throwing an error if no activity.id is set.
+            //         See https://github.com/microsoft/botbuilder-js/issues/1122
+            //      2. By mocking an array of ResourceResponses, we still return the value from `next2()`.
+            //         This prevents any pre-4.6.0 bots with multiple SendActivityHandlers that rely
+            //         on values that are not an array of ResourceResponses.
+            if (!Array.isArray(responses)) {
```

Comment:
why isn't this fix as simple as adding the isArray check on line 71?

Reply:
The reason for the mocked ResourceResponse[] is to make sure that the TranscriptLoggerMiddleware is still able to successfully log the activities . Without an Activity ID the JS TransctipStores would blow up; by creating the mock, we enter the logic where a fake activity is can be generated for the logged activities if

1. The cloned activity does not already have an Id
2. The resourceresponse did not have an Id

Reply:
But if I follow this logic correctly, if someone returns a responses with a length of 1 and activity is length of 2, then you'll still have missing ids, right?